### PR TITLE
Fix integration tests and migrate to test proxy

### PR DESCRIPTION
**Pull request by ostoliarova-msft**
# Description

This pull requests fixes failing tests for SMS SDK.

- As Int cloud is not stable itself, Int tests are removed from the pipelines.
- Several environment variables are renamed to be unique and to avoid shadowing with variables used for other tests.
- Tests are migrated from vcrpy to test proxy.
- New test recordings are created.
- Code refactoring to reduce repeated code.

Additional links that explain test proxy and migration:

[azure-sdk-tools/README.md at main · Azure/azure-sdk-tools (github.com)](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/README.md)

[azure-sdk-for-python/test_proxy_migration_guide.md at main · Azure/azure-sdk-for-python (github.com)](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/test_proxy_migration_guide.md#guide-for-migrating-to-the-test-proxy-from-vcrpy)

[azure-sdk-tools/README.md at main · Azure/azure-sdk-tools (github.com)](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-sms/tests/test_sms_client_e2e.py [hunk 1]
```diff
@@ -101,85 +72,73 @@ def test_send_sms_from_managed_identity(self):
...
             # calling send() with sms values
             sms_client.send(
                 from_="+15550000000",
                 to=[self.phone_number],
                 message="Hello World via SMS")
-        
-        assert str(ex.value.status_code) == "400"
+
+        assert str(
+            ex.value.status_code) == "401"
```
Comment (ostoliarova-msft) [inline sdk/communication/azure-communication-sms/tests/test_sms_client_e2e.py:92]: The service used to return 400 but the validation changed and it's 401 now. The team is aware of this issue and has a ticket for addressing it. Re-record the test if service is re-deployed with a different error handling logic for this endpoint.

File: sdk/communication/azure-communication-sms/tests/test_sms_client_e2e_async.py [hunk 2]
```diff
@@ -106,103 +75,85 @@ async def test_send_sms_from_managed_identity_async(self):
...
                 # calling send() with sms values
                 await sms_client.send(
                     from_="+15550000000",
                     to=[self.phone_number],
                     message="Hello World via SMS")
-        
-        assert str(ex.value.status_code) == "400"
+
+        assert str(
+            ex.value.status_code) == "401"
```
Comment (ostoliarova-msft) [inline sdk/communication/azure-communication-sms/tests/test_sms_client_e2e_async.py:96]: The service used to return 400 but the validation changed and it's 401 now. The team is aware of this issue and has a ticket for addressing it. Re-record the test if service is re-deployed with a different error handling logic for this endpoint.

File: sdk/communication/azure-communication-sms/tests/acs_sms_test_case.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+from devtools_testutils import AzureRecordedTestCase, is_live
+from azure.communication.sms._shared.utils import parse_connection_str
+
+class ACSSMTestCase(AzureRecordedTestCase):
```
Comment (phermanov-msft) [inline sdk/communication/azure-communication-sms/tests/acs_sms_test_case.py:10]: Please fix the typo, I believe it should say ACSSMSTestCase

**Review by phermanov-msft — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-sms/tests/acs_sms_test_case.py (hunk 3 above)
Comment (ostoliarova-msft) [reply sdk/communication/azure-communication-sms/tests/acs_sms_test_case.py]: @phermanov-msft thanks for noticing it. Fixed

**Review by phermanov-msft — APPROVED** [review]

**Review by ilyapaliakou-msft — APPROVED** [review]

**Review by mccoyp — APPROVED** [review]
Thank you so much for your work on this!

**Pull request merged**